### PR TITLE
Allow network configuration via daemon config file.

### LINKED ISSUE #19675: missing options in daemon.conf
Some options seem to be missing from the `daemon.conf` file support in 1.10.0-rc1 - I was trying to set `--bip` and it was ignored for example.

Cross referencing the docs at https://github.com/docker/docker/blob/master/docs/reference/commandline/daemon.md#daemon-configuration-file against the documented options, I think these are missing:

```
  -b, --bridge=""                        Attach containers to a network bridge
  --bip=""                               Specify network bridge IP
  --default-gateway=""                   Container default gateway IPv4 address
  --default-gateway-v6=""                Container default gateway IPv6 address
  -e, --exec-driver="native"             Exec driver to use
  --fixed-cidr=""                        IPv4 subnet for fixed IPs
  --fixed-cidr-v6=""                     IPv6 subnet for fixed IPs
  --icc=true                             Enable inter-container communication
  --insecure-registry=[]                 Enable insecure registry communication
  --ip=0.0.0.0                           Default IP when binding container ports
  --ip-forward=true                      Enable net.ipv4.ip_forward
  --ip-masq=true                         Enable IP masquerading
  --iptables=true                        Enable addition of iptables rules
  --ipv6=false                           Enable IPv6 networking
  --disable-legacy-registry=false        Do not contact legacy registries
  --registry-mirror=[]                   Preferred Docker registry mirror
  --userland-proxy=true        
```

### PR DESCRIPTION
**Pull request by calavera**
Fixes #19675.

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: docs/reference/commandline/daemon.md [hunk 1]
```diff
@@ -859,7 +859,20 @@ This is a full example of the allowed configuration options in the file:
 	"userns-remap": "",
 	"group": "",
 	"cgroup-parent": "",
-	"default-ulimits": {}
+	"default-ulimits": {},
+        "ipv6": false,
```
Comment (tiborvass) [inline docs/reference/commandline/daemon.md]: one too many space for indentation

Comment (calavera) [inline docs/reference/commandline/daemon.md]: fixed

File: daemon/config.go [hunk 2]
```diff
@@ -85,6 +84,7 @@ type CommonConfig struct {
 	// deserialization without the full struct.
 	CommonTLSOptions
 	LogConfig
+	bridgeConfig // bridgeConfig holds bridge network specific configuration.
```
Comment (tiborvass) [inline daemon/config.go]: i wonder if it's gonna get marshalled, because even though it is embedded, it is not exported. Will check.

Comment (tiborvass) [inline daemon/config.go]: Confirmed it works! #TIL

Comment (calavera) [inline daemon/config.go]: I added a test that should tell us :smile:

File: daemon/config_unix.go [hunk 3]
```diff
@@ -37,19 +37,19 @@ type Config struct {
...
-	DefaultIP                   net.IP
-	Iface                       string
-	IP                          string
-	FixedCIDR                   string
-	FixedCIDRv6                 string
-	DefaultGatewayIPv4          net.IP
-	DefaultGatewayIPv6          net.IP
-	InterContainerCommunication bool
+	EnableIPv6                  bool   `json:"ipv6,omitempty"`
+	EnableIPTables              bool   `json:"ip-tables,omitempty"`
```
Comment (tiborvass) [inline daemon/config_unix.go]: `iptables`

Comment (calavera) [inline daemon/config_unix.go]: i-p-t-a-b-l-e-s

**Comment by jfrazelle** [comment]
LGTM

**Comment by jfrazelle** [comment]
huh
```
# github.com/docker/docker/docker
22:42:40 docker/daemon_unix_test.go:9:2: cannot find package "github.com/crosbymichael/docker/daemon" in any of:
22:42:40 	/usr/local/go/src/github.com/crosbymichael/docker/daemon (from $GOROOT)
22:42:40 	/go/src/github.com/crosbymichael/docker/daemon (from $GOPATH)
22:42:40 	/go/src/github.com/docker/docker/vendor/src/github.com/crosbymichael/docker/daemon
22:42:40 FAIL	github.com/docker/docker/docker [setup failed]
22:42:56 ok  	github.com/docker/docker/api	0.012s	coverage: 92.5% of statements
```

File: docs/reference/commandline/daemon.md [hunk 4]
```diff
@@ -859,7 +859,20 @@ This is a full example of the allowed configuration options in the file:
 	"userns-remap": "",
 	"group": "",
 	"cgroup-parent": "",
-	"default-ulimits": {}
+	"default-ulimits": {},
+       "ipv6": false,
+       "ip-tables": false,
```
Comment (tiborvass) [inline docs/reference/commandline/daemon.md]: `iptables`

File: docker/daemon_unix_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// +build daemon,!windows
+
+package main
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/crosbymichael/docker/daemon"
```
Comment (jfrazelle) [inline docker/daemon_unix_test.go]: this lol

File: docs/reference/commandline/daemon.md (hunk 4 above)
Comment (tiborvass) [inline docs/reference/commandline/daemon.md]: this is your new nightmare lol... IPTABLES IPTABLES IPTABLES!!

Comment (calavera) [inline docs/reference/commandline/daemon.md]: STAP!!!

**Comment by calavera** [comment]
@jfrazelle goimports freaked out for a moment.

**Comment by jfrazelle** [comment]
i figured ;) LGTM

File: docker/daemon_unix_test.go (hunk 5 above)
Comment (tiborvass) [inline docker/daemon_unix_test.go]: lol that's some funny `goimports`, it's because `crosbymichael` is before `docker` lexicographically :D

**Comment by tiborvass** [comment]
LGTM

**Comment by thaJeztah** [comment]
It's greeeeeeen \o/ :frog:

**Comment by jhowardmsft** [comment]
@calavera Congrats on reaching #300 :whale: :smile:

**Comment by thaJeztah** [comment]
Woohoo, congrats @calavera ! :tada:

**Pull request merged**